File: webApp/src/main/java/com/builtin/webapp/dto/UserDto.java
package com.builtin.webapp.dto;

import lombok.Data;

@Data
public class UserDto {
    private Long id;
    private String firstName;
    private String lastName;
    private String email;
    private String userType;
    private Long providerId;

    public boolean canAddItems() {
        return "PROVIDER".equals(userType) || "SUPER_ADMIN".equals(userType);
    }

    public boolean canEdit(com.builtin.webapp.dto.ItemDto item) {
        if ("SUPER_ADMIN".equals(userType)) return true;
        if ("PROVIDER".equals(userType) && providerId != null && item.getProvider() != null) {
            return providerId.equals(item.getProvider().getId());
        }
        return false;
    }

    public String getDisplayName() {
        return firstName + " " + lastName;
    }
}

File: iosApp/BuiltIn/Models/UserDto.swift
import Foundation

struct UserDto: Codable, Equatable {
    let id: Int
    let firstName: String
    let lastName: String
    let email: String
    let userType: UserRole
    let providerId: Int?
    let deliveryAccountId: Int?

    var displayName: String { "\(firstName) \(lastName)" }

    var canAddItems: Bool {
        userType == .provider || userType == .superAdmin
    }

    func canEdit(item: ItemDto) -> Bool {
        if userType == .superAdmin { return true }
        if userType == .provider, let pid = providerId, let provider = item.provider {
            return pid == provider.id
        }
        return false
    }
}

enum UserRole: String, Codable, Equatable {
    case customer       = "CUSTOMER"
    case provider       = "PROVIDER"
    case superAdmin     = "SUPER_ADMIN"
    case enterpriseUser = "ENTERPRISE_USER"
    case delivery       = "DELIVERY"
}

// MARK: - Login

struct LoginRequest: Encodable {
    let username: String
    let password: String
}
